scratch_space/README.md

```diff
@@ -0,0 +1,55 @@
+# [![Build Status](https://travis-ci.org/dart-lang/build.svg?branch=master)](https://travis-ci.org/dart-lang/build)
+
+## [`ScratchSpace`][dartdoc:ScratchSpace]
+
+A `ScratchSpace` is a thin wrapper around a temporary directory. The
+constructor takes zero arguments, so making one is as simple as doing
+`new ScratchSpace()`.
+
+### Adding assets to a `ScratchSpace`
+
+To add assets to the `ScratchSpace`, you use the `ensureAssets` method, which
+takes an `Iterable<AssetId>` and an `AssetReader` (for which you should
+generally pass your `BuildStep` which implements that interface).
+
+**Note:** It is important to note that the `ScratchSpace` does not guarantee
+that the version of a file within it is the most updated version, only that
+some version of it exists. For this reason you should create a new
+`ScratchSpace` for each build (within a single build files cannot change, but
+between builds they might).
+
+### Deleting a `ScratchSpace`
+
+When you are done with a `ScratchSpace` you should call `delete` on it to make
+sure it gets cleaned up, otherwise you can end up filling up the users tmp
+directory.
+
+**Note:** You cannot delete individual assets from a `ScratchSpace` today, you
+can only delete the entire thing. If you have a use case for deleting
+individual files you can [file an issue][tracker]. 
+
+### Getting the actual File objects for a `ScratchSpace`
+
+When invoking an external binary, you probably need to tell it where to look
+for files. There are a few fields/methods to help you do this:
+
+  * `String get packagesDir`: The `packages` directory in the `ScratchSpace`.
```

Comment:
Will we write a `.packages` as well? Do we expect tools to continue to understand the directory style in addition to the manifest style?

Reply:
We could certainly... we would have to update it any time you call `ensureAssets` but that wouldn't be difficult.

DDC/Analyzer don't need it though, so I think maybe we could wait and add it as an option later on if somebody asks for it?

Reply:
Makes sense